### ISSUE TITLE
with-loading example crashes with the new EventEmitter refactor

### ISSUE THREAD

**Issue by jasonkylefrank** [zeit/next.js#4982]
# Examples bug report

## Example name
with-loading

## Describe the bug
Running this example app causes an "internal server error" (500).  Reverting to `Router.onRouteChangeStart`, `Router.onRouteChangeComplete`, `Router.onRouteChangeError`  solves the crash (these methods were removed in #4864, causing this crash).  Another way to solve it is to change  the `next` version in the example's `package.json` from `6.1.1` to `7.0.0-canary.0`  

## To Reproduce
Steps to reproduce the behavior, please provide code snippets or a repository:
1. Follow the "how to use" instructions on the [example's readme](https://github.com/zeit/next.js/tree/canary/examples/with-loading).
2. Run the app via `npm run dev` and load a browser port 3000 to connect to it.
3. The terminal shows it then builds page `/`
4. See error in browser: "Internal server error".
5. See the error in the terminal:  `Cannot read property 'on' of undefined` (pointing to the line with `Router.events.on('routeChangeStart', (url) => { ... `

## Expected behavior
The app should load without crashing on the server.

## Screenshots
![image](https://user-images.githubusercontent.com/7502365/44303194-4d7e7880-a308-11e8-82a5-7676053e46ac.png)

![image](https://user-images.githubusercontent.com/7502365/44303199-6d15a100-a308-11e8-837c-9a18c1c805f2.png)

![image](https://user-images.githubusercontent.com/7502365/44303201-8a4a6f80-a308-11e8-9f31-3f198be03119.png)

## System information
 - OS: macOS
 - Browser: Chrome
 - Version of Next.js: 6.1.1 (that's what is listed in `package.json` for this example)

## Additional context
I assume the ideal way to solve this is to update the version of `next` in the example's `package.json`?  I may submit a pull request for that and reference this issue.

**Comment by timneutkens** [comment]
> I assume the ideal way to solve this is to update the version of next in the example's package.json? I may submit a pull request for that and reference this issue.

Yeah lets upgrade to canary 👍

**Comment by jasonkylefrank** [comment]
@timneutkens Actually, it looks like the problem may be with the `create-next-app`, as it relates to creating the example.  

In the _actual repo code_ (not the code that gets generated by `create-next-app`) I just noticed that in the example's `package.json`, the `next` version is already set to `latest`.  

But in the `package.json` that comes from:
`npx create-next-app --example with-loading with-loading-app`,
the `package.json` has the `next` version set to `6.1.1`, which was causing the crash.

So at a minimum, there appears to be a problem with `create-next-app`.  But I don't how to fix that.

--------

Additionally, I tried running the example by drilling into the actual repo's example directory and doing an `npm install` and then running that example.  But that has another problem:

![image](https://user-images.githubusercontent.com/7502365/44303371-460d9e00-a30d-11e8-982c-342b9c97e466.png)

Not sure if this is to be expected because I don't know if we should be able to just run these examples like this.  😀

**Comment by timneutkens** [comment]
The latest version of create-next-app should install from the `canary` branch.

**Comment by jasonkylefrank** [comment]
Hey @timneutkens, I was using the `npx create-next-app ...`, per the example's readme.  But just to see if it would make any difference, I just installed `create-next-app` globally and then retried the example.  But the same thing happened. 😃  But maybe I'm still doing something wrong?

**Comment by Raikhen** [comment]
Mmm, I don't think that the bug is related to `create-next-app`, given that if you make a manual download the issue persists. Also, it throws me the error with `"next": "latest"` in `"dependencies"`.

**Comment by radeno** [comment]
@jasonkylefrank do you have in package.json next version `canary`?
@Raikhen `latest` version is always stable. So for now 6.1.1. Use `canary`
@timneutkens latest version of `create-next-app` doesnt install canary version. it uses npm or yarn command to install package without version argument. so it is always last stable. https://github.com/segmentio/create-next-app/blob/master/lib/utils/install.js#L44 `packages` argument is simple array https://github.com/segmentio/create-next-app/blob/master/lib/utils/install.js#L44

**Comment by timneutkens** [comment]
I wonder if this PR wasn't enough then: https://github.com/segmentio/create-next-app/pull/36/files

I'm not saying create-next-app should install canary, it should respect the package.json from the example 🕵️

**Comment by radeno** [comment]
If i understand it right It only downloads NextJs from Github (canary branch), unzip it and copy requested example directory content.

**Comment by jasonkylefrank** [comment]
UPDATE: I'm not sure what changed, but today this issue appears to be **fixed**?  
**Can you guys confirm?**

I ran `npx create-next-app --example with-loading with-loading-app` and it ran fine without crashing.  Also, unlike before, it now installs with `7.0.1` as the `next` version:

![image](https://user-images.githubusercontent.com/7502365/46431194-285b9300-c719-11e8-927c-bbe978663d94.png)

What changed?

**Comment by jasonkylefrank** [comment]
I think maybe I've "put 2 and 2 together" to figure out why the original problem existed - where `create-next-app` listed the `next` dependency version as `6.1.1` (which ended up breaking things).  Feel free to correct me on this, I'd love to learn if this understanding is flawed.  😀

First, I noticed that the `with-loading` example has had its `next` dependency version set to **`latest`** for a long time (it hasn't changed for a long time):

![image](https://user-images.githubusercontent.com/7502365/46443354-273b5d80-c73b-11e8-94c2-c35614ec51fb.png)

Next, I examined `create-next-app`'s source code better to see what it downloads and does.  After it downloads the example, it simply runs `npm install` (or the equivelant `yarn` command) from the example's directory.  Which is to say, it installs the **`latest`** version of `next`.    

This last part is a conjecture on my part.  But I think what happened is that the Zeit folks push (i.e., `npm publish`) the _master_ branch code to NPM.  And furthermore I'm guessing they only publish the master branch to NPM when it's `package.json` lists a version which does **not** have "-canary" in it.  And judging by the history of `package.json` on the master branch, I'd say that `6.1.1` was the version that meets this criteria at that time.  (They had committed some versions like `6.1.1-canary.0` afterward but I'm assuming they would not publish that to NPM)

![image](https://user-images.githubusercontent.com/7502365/46443868-65d21780-c73d-11e8-8e74-1ce9ef138bad.png)

Lastly, let's take a look at the screenshot I originally shared, where I spotted the difference between the `next` version that I expected to be installed and the `next` version which actually installed.   You'll notice I was viewing the `package.json` for the **canary** branch.   But as I said above, I'm guessing that the _master_ branch which would have actually been published to _**NPM**_ at that time would have had version `6.1.1` listed in its `package.json`.  As such, that would have been the version that `create-next-app` would have installed for the example - causing the crash that I originally showed.

![image](https://user-images.githubusercontent.com/7502365/46443111-25bd6580-c73a-11e8-8efa-04e86ed69c2a.png)

Again, chime-in if this is not correct. 😀  I thought I'd capture this for my own learning sake, if nothing else.  Maybe someone else can benefit from it as well.  
@timneutkens
